### PR TITLE
Core/Spellsystem: Implemented burning embers system

### PR DESCRIPTION
**Pull request by F03SD**
Thanks @Shauren to point.

### CODE REVIEW

File: src/server/game/Entities/Unit/Unit.cpp [hunk 1]
```diff
@@ -6379,6 +6379,30 @@ bool Unit::HandleAuraProc(Unit* victim, uint32 /*damage*/, Aura* triggeredByAura
             }
             break;
         }
+        case SPELLFAMILY_WARLOCK:
+        {
+            switch (dummySpell->Id)
+            {
+                // Burning embers spell proc
+                case 108647:
```
Comment (DDuarte) [inline src/server/game/Entities/Unit/Unit.cpp]: Isn't there a way to do this without hardcoding all these spell ids?

File: src/server/game/Entities/Unit/Unit.cpp [hunk 2]
```diff
@@ -6379,6 +6379,30 @@ bool Unit::HandleAuraProc(Unit* victim, uint32 /*damage*/, Aura* triggeredByAura
...
+                {
+                    bool isCrit = (procEx & PROC_EX_CRITICAL_HIT);
+
+                    // Increase burning emberse
+                    if (procSpell->Id == 29722) // Incinerate
+                        SetPower(POWER_BURNING_EMBERS, GetPower(POWER_BURNING_EMBERS)+(isCrit ? 2 : 1));
```
Comment (DDuarte) [inline src/server/game/Entities/Unit/Unit.cpp]: `ModifyPower(POWER_BURNING_EMBERS, isCrit ? 2 : 1);`

File: src/server/game/Entities/Unit/Unit.cpp (hunk 1 above)
Comment (F03SD) [inline src/server/game/Entities/Unit/Unit.cpp]: @DDuarte up to me cases  hardcode too. I just thought ...

**Comment by Shauren** [comment]
I have not looked over this code much but one thing is certain - its a proc aura so it must be implemented as aurascript with proc hook, not in Unit::HandleAuraProc

**Comment by F03SD** [comment]
@Shauren hmm , ty for the point , close this PR i will make new with new implemented aurascript within a few days

**Comment by DDuarte** [comment]
@F03SD: do the changes on top of this PR, there's no need to open another :)
Thanks

**Comment by F03SD** [comment]
@DDuarte kay.

**Comment by F03SD** [comment]
@Shauren  @DDuarte done

**Comment by P-Kito** [comment]
How do you know 1 ember is 10 small embers?

**Comment by F03SD** [comment]
@P-Kito yea

**Pull request closed without merging**